### PR TITLE
MESH-1621/Script to create snapshot of all user balances 

### PR DESCRIPTION
**Pull request by JMoore96**
Created a script to get users balances and store them in a CSV file.

To run: enter `yarn polyx_balances` from cli folder

### modified logic

- added userBalancePOLYX.js file

### CODE REVIEW

File: scripts/cli/util/userBalancePOLYX.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+				miscFrozen: JSON.stringify(skAccountData.data.miscFrozen),
+			});
+		}
+
+		let did = await api.query.identity.keyToIdentityIds(pk);
+		let accountData = await api.query.system.account(pk);
+
+		if (did.toString() != empty_did) {
+			data.push({
+				AccountId: pk,
```
Comment (adamdossa) [inline scripts/cli/util/userBalancePOLYX.js:58]: I'd suggest adding an additional field:
`isPrimaryKey`
which is true / false depending on whether the AccountId is the primary key or a secondary key for the DID.

**Comment by adamdossa** [comment]
@JMoore96 left a couple of minor comments. Please can we also remove:
`scripts/cli/util/usersPolyxBalance.csv`
from the committed files.

**Review by CJP10 — APPROVED** [review]
LGTM! There seems to be some overlap with the other script, maybe we share some code to avoid duplication.

File: scripts/cli/util/userBalancePOLYX.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		let listOfSecondaryKeyObjects = listOfDids[i][1]["secondary_keys"];
+
+		for (let j = 0; j < listOfSecondaryKeyObjects.length; j++) {
+			// secondary key
+			let sk = JSON.parse(listOfSecondaryKeyObjects[j]["signer"]).Account;
+			// get sk did
+			let skDid = await api.query.identity.keyToIdentityIds(sk);
+			// getting secondary key balance info
+			let skAccountData = await api.query.system.account(sk);
+			data.push({
```
[the remark this answers is not in the document]
Comment (JMoore96) [reply scripts/cli/util/userBalancePOLYX.js]: understood, made the changes.

File: scripts/cli/util/userBalancePOLYX.js (hunk 1 above)
Comment (JMoore96) [reply scripts/cli/util/userBalancePOLYX.js]: I added the new field in my latest commit.

**Review by adamdossa — APPROVED** [review]

**Comment by F-OBrien** [comment]
Late comment, and I assume this is a known limitation of the script but I thought I'd mention it anyway. 
This script will not return the balance of any key that has been unlinked. So if for some reason a user removes a key from their identity records while it still contains POLYX it will not be reported. I don't know the intended use of this script but it might be worth including a note to that effect in it.

**Comment by adamdossa** [comment]
@F-OBrien thanks - this script is just for us to gauge whether users are getting close to running out of ITN POLYX at a high-level rather than for detailed analysis - as you say there are some edge cases it misses. Our thinking is that as the ITN challenges progress we might need to "top up" ITN POLYX balances, and this script just helps us monitor that situation.

**Pull request merged**